cpp/tests/utils/env.hpp

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <string>
+#include <unordered_map>
+
+namespace kvikio::test {
+/**
+ * @brief RAII class used to temporarily set environment variables to new values upon construction,
+ * and restore them to previous values upon destruction.
+ */
+class EnvVarContext {
```

Comment:
neat! I might borrow this for cudf compression tests, they are getting to a point where more than one env var is being set

Reply:
IMO it's fine in tests, it's not a part of the functionality of the kvikIO library